packages/native/src/components/MapView/MapMarker.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+import { LatLng } from "react-native-maps";
+import { MapMarkerProps } from "./types";
+import { Marker as WebMarker, MarkerProps } from "@react-google-maps/api";
+interface IMarkerContext {
+  pinColor?: string | undefined;
+  calloutOpened: boolean;
+  position: LatLng | undefined;
+  toggleCallout: (value: boolean) => void;
+}
+
+export const markerContext = React.createContext<IMarkerContext>({
+  pinColor: "#E74C3C",
+  calloutOpened: false,
+  position: undefined,
+  toggleCallout: () => {},
+});
+
+const MapMarker: React.FC<MapMarkerProps> = ({
+  pinColor,
+  latitude,
+  longitude,
+  title,
+  description,
+  children,
+}) => {
+  const { Provider } = markerContext;
+  const [calloutOpened, toggleCallout] = React.useState(false);
+  const [marker, setMarker] = React.useState<MarkerProps | null>(null);
+  const handleMarkerClick = () => toggleCallout(true);
+  const handleOnLoad = (m: MarkerProps) => setMarker(m);
+  const mappedChildren = React.Children.map(children, (child, index) => {
+    return React.cloneElement(child as React.ReactElement, {
+      index,
+      anchor: marker,
+    });
+  });
+  return (
```

Comment:
We generally prefer newlines before `return` for readability
```suggestion

  return (
```